### PR TITLE
DevWs for RoomsBeautiful containing containing Curtain

### PR DESCRIPTION
**Pull request by devropes**

### CODE REVIEW

**Comment by shrakuma** [comment]
@MAblewiczS can you please review this. Also, this is a previously created DTH and this custom DTH should be safe and only used for Rooms beautiful devices cc @tpmanley @greens  please share your opinion about reviewing this DTH. As its not a new custom DTH and was once approved via the new submission process, I assume that the changes made here should be reviewable.

File: devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy [hunk 1]
```diff
@@ -21,16 +21,18 @@ metadata {
 		capability "Battery"
 		capability "Configuration"
 		capability "Refresh"
-		capability "Window Shade"
-		capability "Health Check"
+        capability "Health Check"
```
Comment (tpmanley) [inline devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy:24]: nit: please use tabs for indentation in the whole file

File: devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy [hunk 2]
```diff
@@ -21,16 +21,18 @@ metadata {
...
+		capability "Window Shade"		
+        capability "Switch"
+        capability "Switch Level"
 
 		attribute("replay", "enum")
 		attribute("battLife", "enum")
 
-		//command "pause"
+		command "pause"
```
Comment (tpmanley) [inline devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy]: `pause` is a command in the `Window Shade` capability so this custom command should be removed

File: devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy [hunk 3]
```diff
@@ -21,16 +21,18 @@ metadata {
 		capability "Battery"
 		capability "Configuration"
 		capability "Refresh"
-		capability "Window Shade"
-		capability "Health Check"
+        capability "Health Check"
+		capability "Window Shade"		
+        capability "Switch"
+        capability "Switch Level"
```
Comment (tpmanley) [inline devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy:27]: Just so you know, we're in the process of adding a new `Window Shade Level` capability which will be preferred over `Switch Level` for devices like this. 

@dkirker do you think we're ready to use that new capability right now or should they stick with `Switch Level` for the moment?

File: devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy [hunk 4]
```diff
@@ -128,31 +150,60 @@ def parse(String description) {
 	}
 }
 
+def off() {
+	zigbee.off() +
+    sendEvent(name: "level", value: 0)
```
Comment (tpmanley) [inline devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy]: Are these events necessary? Typically we let the device report its new state and then generate the appropriate events.

File: devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy [hunk 5]
```diff
@@ -128,31 +150,60 @@ def parse(String description) {
...
+		time = now()
+	}
+	else{
+		time = now() - state.updatedDate
+	}
+	state.updatedDate = now()
+    log.trace ("Time: ${time}")
+        
+    if (time > 1000 ){
+        zigbee.command(0x0102, 0x05, zigbee.convertToHexString(100-value, 2)) +
```
Comment (tpmanley) [inline devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy]: I suggest defining these two functions in the DTH:
```
private getCLUSTER_WINDOW_COVERING() { 0x0102 }
private getCOMMAND_GOTO_LIFT_PERCENTAGE() { 0x05 }
```

Then here you can write: `zigbee.command(CLUSTER_WINDOW_COVERING, COMMAND_GOTO_LIFT_PERCENTAGE, zigbee.convertToHexString(100-value, 2))`. The benefit is it is easier to understand what it is doing.

File: devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy [hunk 6]
```diff
@@ -128,31 +150,60 @@ def parse(String description) {
...
+        sendEvent(name: "level", value: value) +
```
Comment (tpmanley) [inline devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy]: I'm not sure if this works to append the `log.debug` to the return value but I would suggest logging the message first and then returning only the `zigbee.command` and the `sendEvent`.

File: devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy [hunk 7]
```diff
@@ -167,93 +218,76 @@ private handleBatteryEvent(volts) {
...
-			log.debug "${linkText} - Batt: ${value} **** Volts: ${volts/10}v **** Percent: ${percent}%"
+			log.info "${linkText} - Batt: ${value} **** Volts: ${volts/10}v **** Percent: ${percent}%"
 		}
 	}
 }
 
 def refresh() {
 	zigbee.onOffRefresh() +
-	zigbee.readAttribute(0x0001, 0x0020) +
+    zigbee.readAttribute(0x0102, 0x0008) 	// Window Lift Percentage Attribute
```
Comment (tpmanley) [inline devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy]: Missing the `+`

File: devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy [hunk 8]
```diff
@@ -167,93 +218,76 @@ private handleBatteryEvent(volts) {
...
+	zigbee.readAttribute(0x0001, 0x0020) +	// Battery Voltage Attribute
```
Comment (tpmanley) [inline devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy]: Better to use `zigbee.POWER_CONFIGURATION_CLUSTER` than the hardcoded cluster ID

File: devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy (hunk 3 above)
Comment (devropes) [reply devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy:27]: I would still like to use Switch Level as it is compatible with Alexa.

File: devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy (hunk 2 above)
Comment (devropes) [reply devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy]: I have now fixed this.

File: devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy (hunk 4 above)
Comment (devropes) [reply devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy]: This is necessary for my device as it doesn't send a level event when it is told to fully open or close.

File: devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy (hunk 5 above)
Comment (devropes) [reply devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy]: I have now fixed this.

File: devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy (hunk 6 above)
Comment (devropes) [reply devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy]: I have now fixed this.

File: devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy (hunk 7 above)
Comment (devropes) [reply devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy]: I have now fixed this.

File: devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy (hunk 8 above)
Comment (devropes) [reply devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy]: I have now fixed this.

File: devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy (hunk 1 above)
Comment (devropes) [reply devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy:24]: In the IDE, the indentation is correct, but when I commit it to Github, the indentation gets messed up. I've tried to fix this before, but it doesn't seem to work.

Comment (tpmanley) [reply devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy:24]: Unfortunately it's tough to get the indentation right when using our IDE. I suggest copying the code into a different editor and fixing it up there.

File: devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy (hunk 3 above)
Comment (tpmanley) [reply devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy:27]: I have filed a request to make the Alexa integration work with `Window Shade` and `Window Shade Level` but this is fine for now.

File: devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy (hunk 1 above)
Comment (devropes) [reply devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy:24]: I have now fixed this.

**Review by tpmanley — APPROVED** [review]

**Comment by tpmanley** [comment]
Thanks for making the changes @devropes . I pointed the PR against the `staging` branch so that it goes to production next Tuesday.

File: devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy (hunk 3 above)
Comment (dkirker) [reply devicetypes/rooms-beautiful/rooms-beautiful-curtain.src/rooms-beautiful-curtain.groovy:27]: @tpmanley Window Shade Level isn't quite ready. I think @larsfinander did add it, at least Window Shade, to the voice integration stuff. I'll put that on the tracker I have been using for Window Shades.

**Review by dkirker — COMMENTED** [review]
Left a comment inline re Window Shade Level that @tpmanley was asking about.

**Pull request merged**